pkgs/servers/http/unit/unit-rootless.patch

```diff
@@ -0,0 +1,24 @@
+diff --git a/src/nxt_process.c b/src/nxt_process.c
+index 5952029..7b29369 100644
+--- a/src/nxt_process.c
++++ b/src/nxt_process.c
+@@ -136,7 +136,7 @@ nxt_process_start(nxt_task_t *task, nxt_process_t *process)
+
+     nxt_random_init(&thread->random);
+
+-    if (init->user_cred != NULL && getuid() == 0) {
++    if (init->user_cred != NULL) {
```

Comment:
Why is this OK? Do you have a blog post from the developers explaining this? Will the software run as expected if you make this change? Some sort of explanation of proof of why this is OK would be appreciated.

Reply:
Taken from here - https://github.com/nginx/unit/issues/228

Reply:
:+1: great! Could you please add a link to that issue in the source code with a note that the patch should be removed once the code has been merged upstream?

Reply:
Not merged to upstream, temporary solution. Probably the developers then finalize

Reply:
I understand it is not merged upstream. The developers will eventually merge this code, or code similar to this upstream. When the developers merge that code upstream then this patch should be removed. If you do not include a link to that issue in the package file it is possible when an update happens that could get missed.

Please include link to the issue for tracking purposes.

Reply:
@Izorkin You marked this as resolved, but I don't think I see the comment anywhere. Can you point me to it?

Reply:
About the use of the patch wrote in the comments. I checked the service on my server - it works

Reply:
```
● unit.service - Unit App Server
   Loaded: loaded (/nix/store/997ams0c3w3r0x8v1mgsmrakq7ai6w9q-unit-unit.service/unit.service; enabled; vendor preset: enabled)
   Active: active (running) since Fri 2019-03-15 22:26:42 MSK; 20h ago
 Main PID: 7854 (unitd)
    Tasks: 12 (limit: 1172)
   Memory: 114.3M
   CGroup: /system.slice/unit.service
           ├─7854 unit: main v1.8.0 [/nix/store/c7469fsxzhwzjh0khkamrx3zbcx2dzn5-unit-1.8.0/bin/unitd --control unix:/run/unit/control.unit.sock --pid /run/unit/unit.pid --log /var/log/unit/unit.log --state /var/spool/unit --no-daemon --user unit --group unit]
           ├─7879 unit: controller
           ├─7880 unit: router
           ├─7882 unit: "php-72-roundcube" application
           ├─7898 unit: "php-72-roundcube" application
           ├─7903 unit: "php-72-roundcube" application
           ├─7904 unit: "php-72-roundcube" application
           ├─7907 unit: "php-72-info" application
           ├─7918 unit: "php-72-bench" application
           └─7928 unit: "php-72-prober" application

мар 15 22:26:41 web.elven.pw systemd[1]: Starting Unit App Server...
мар 15 22:26:41 web.elven.pw unitd[7854]: 2019/03/15 22:26:41 [info] 7854#7854 unit started
мар 15 22:26:41 web.elven.pw niavz4asw6q16r82kg115ak05hpw6ak6-unit-script-unit-post-start[7855]:   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
мар 15 22:26:41 web.elven.pw niavz4asw6q16r82kg115ak05hpw6ak6-unit-script-unit-post-start[7855]:                                  Dload  Upload   Total   Spent    Left  Speed
мар 15 22:26:42 web.elven.pw niavz4asw6q16r82kg115ak05hpw6ak6-unit-script-unit-post-start[7855]: [237B blob data]
мар 15 22:26:42 web.elven.pw niavz4asw6q16r82kg115ak05hpw6ak6-unit-script-unit-post-start[7855]: {
мар 15 22:26:42 web.elven.pw niavz4asw6q16r82kg115ak05hpw6ak6-unit-script-unit-post-start[7855]:         "success": "Reconfiguration done."
мар 15 22:26:42 web.elven.pw niavz4asw6q16r82kg115ak05hpw6ak6-unit-script-unit-post-start[7855]: }
мар 15 22:26:42 web.elven.pw systemd[1]: Started Unit App Server.

```

Reply:
put a link to that nginx/unit issue near the patches

Reply:
Thank you!